Investigations/SUPPORT_Notebook.ipynb

```diff
@@ -0,0 +1,440 @@
+    "\n",
+    "synthetic_sample = vae.generate(data_supp.shape[0])\n",
+    "\n",
+    "if torch.cuda.is_available():\n",
+    "    synthetic_sample = pd.DataFrame(\n",
+    "        synthetic_sample.cpu().detach(), columns=reordered_dataframe_columns\n",
+    "    )\n",
+    "else:\n",
+    "    synthetic_sample = pd.DataFrame(\n",
+    "        synthetic_sample.detach(), columns=reordered_dataframe_columns\n",
+    "    )\n",
+    "\n",
+    "# Reverse the transformations\n",
+    "\n",
+    "synthetic_supp = reverse_transformers(\n",
+    "    synthetic_set=synthetic_sample,\n",
+    "    data_supp_columns=data_supp.columns,\n",
+    "    cont_transformers=continuous_transformers,\n",
+    "    cat_transformers=categorical_transformers,\n",
+    "    pre_proc_method=pre_proc_method,\n",
+    ")\n",
+    "\n"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "7a7d8d8c",
+   "metadata": {},
+   "source": [
+    "## Synthetic Variable Visualisation\n",
+    "\n",
+    "Here we want to visualise the synthetic variables generated and compare them to the original set"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "7fdaf4f1",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "#%% -------- Plot Histograms For All The Variable Distributions -------- #\n",
+    "\n",
+    "plot_variable_distributions(\n",
+    "    categorical_columns=original_categorical_columns,\n",
+    "    continuous_columns=original_continuous_columns,\n",
+    "    data_supp=data_supp,\n",
+    "    synthetic_supp=synthetic_supp,\n",
+    "    saving_filepath=\"\",\n",
+    "    pre_proc_method=pre_proc_method,\n",
+    ")\n"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "09630b17",
+   "metadata": {},
+   "source": [
+    "## Metric evaluation\n",
+    "\n",
+    "We use the SDV evaluation framework. Supply the metrics you wish to find in the user_metrics list from SDV guidance. Can start here: https://sdv.dev/SDV/user_guides/evaluation/single_table_metrics.html\n",
+    "\n",
+    "Note that not all of these will work, some are hit and miss. We predominantly rely on continuous and discrete KL divergence measures. You can also input <b>\"gower\"</b> and this will calculate the gower distance using the gower library."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "0f240e21",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "#%% -------- SDV Metrics -------- #\n",
+    "\n",
+    "# Define the metrics you want the model to evaluate\n",
+    "\n",
+    "gower = False\n",
+    "\n",
+    "metrics = distribution_metrics(\n",
+    "    gower_bool=gower,\n",
+    "    data_supp=data_supp,\n",
+    "    synthetic_supp=synthetic_supp,\n",
+    "    categorical_columns=original_categorical_columns,\n",
+    "    continuous_columns=original_continuous_columns,\n",
+    "    saving_filepath=None,\n",
+    "    pre_proc_method=pre_proc_method,\n",
+    ")\n"
```

Comment:
Possibly add a print of the metrics here to show the outputs in the notebook as well